### PR TITLE
Don't check in zstd compressed files

### PR DESCRIPTION
**Pull request by ripatel-fd**
Compress gui assets on startup to reduce git bloat.

### CODE REVIEW

File: src/app/fdctl/run/tiles/fd_gui.c [hunk 1]
```diff
@@ -268,6 +269,30 @@ privileged_init( fd_topo_t *      topo,
...
+    char const * ext = strrchr( f->name, '.' );
+    if( !ext ) continue;
+    if( !strcmp( ext, ".html" ) ||
+        !strcmp( ext, ".css"  ) ||
+        !strcmp( ext, ".js"   ) ||
+        !strcmp( ext, ".svg"  ) ) {}
+    else continue;
+
+    ulong   zstd_bufsz = *f->data_len;
+    uchar * zstd_buf   = malloc( zstd_bufsz );
```
Comment (ripatel-fd) [inline src/app/fdctl/run/tiles/fd_gui.c:285]: @mmcgee-jump I know usage of `malloc` is generally frowned upon. 
But this is one-time initialization code, so I think it gets a pass.

Comment (ripatel-fd) [reply src/app/fdctl/run/tiles/fd_gui.c:285]: Would still be nice to have these files huge page backed

Comment (mmcgee-jump) [reply src/app/fdctl/run/tiles/fd_gui.c:285]: I don't like using ZSTD in privileged init here. Can we move this code to `unprivileged_init`? Also you can just do `fd_wksp_alloc( fd_wksp_containing( gui_ctx ), zstd_bufsz )` to get the memory in the GUI gigantic pages. You could also set `loose_sz` to some minimum value to ensure these allocs succeed.

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/app/fdctl/run/tiles/fd_gui.c (hunk 1 above)
Comment (ripatel-fd) [reply src/app/fdctl/run/tiles/fd_gui.c]: done

File: src/app/fdctl/run/tiles/fd_gui.c [hunk 2]
```diff
@@ -270,6 +288,51 @@ privileged_init( fd_topo_t *      topo,
...
+    if( !ext ) continue;
+    if( !strcmp( ext, ".html" ) ||
+        !strcmp( ext, ".css"  ) ||
+        !strcmp( ext, ".js"   ) ||
+        !strcmp( ext, ".svg"  ) ) {}
+    else continue;
+
+    ulong   zstd_bufsz = fd_spad_alloc_max( spad, 1UL );
+    uchar * zstd_buf   = fd_spad_prepare( spad, 1UL, zstd_bufsz );
+    ulong zstd_sz = ZSTD_compress( zstd_buf, zstd_bufsz, f->data, *f->data_len, 19 );
```
Comment (ripatel-fd) [inline src/app/fdctl/run/tiles/fd_gui.c:321]: This seems to do malloc() under the hood. Need to switch to the static compression API

File: src/app/fdctl/run/tiles/fd_gui.c (hunk 1 above)
Comment (ripatel-fd) [reply src/app/fdctl/run/tiles/fd_gui.c]: @mmcgee-jump I'm working on the change now. It has about 100 LOC additional code bloat.

File: src/app/fdctl/run/tiles/fd_gui.c (hunk 2 above)
Comment (ripatel-fd) [reply src/app/fdctl/run/tiles/fd_gui.c]: Fixed

**Review by mmcgee-jump — DISMISSED** [review]

**Comment by ptaffet-jump** [comment]
What about compressing them at compile time? Would that require another dependency?

**Review by mmcgee-jump — DISMISSED** [review]

**Review by riptl — APPROVED** [review]

**Pull request merged**